SPDX-FileCopyrightText: 2021 fosslinux <fosslinux@aussies.space>

SPDX-License-Identifier: GPL-2.0-or-later

We disable the following programs for the following reasons:

* script: doesn't build with musl.
* flock: GCC doesn't like some directive.
* fstrim: ??? doesn't build (missing/bad linux header?)
* fsfreeze: ditto.
* hexdump: musl incompatibility.
* column: musl incompatibility.

diff --color -ru misc-utils/Makefile.am misc-utils/Makefile.am
--- util-linux-2.19.1/misc-utils/Makefile.am	2021-06-13 09:34:49.669370352 +1000
+++ util-linux-2.19.1/misc-utils/Makefile.am	2021-06-13 09:42:06.644523490 +1000
@@ -6,8 +6,8 @@
 sbin_PROGRAMS =
 usrsbin_exec_PROGRAMS =
 
-usrbin_exec_PROGRAMS = cal ddate logger look mcookie \
-	namei script whereis scriptreplay
+usrbin_exec_PROGRAMS = cal ddate look mcookie \
+	namei whereis scriptreplay
 EXTRA_DIST += README.cal README.ddate README.namei README.namei2
 
 mcookie_SOURCES = mcookie.c $(top_srcdir)/lib/md5.c
@@ -50,18 +50,6 @@
 wipefs_LDADD = $(ul_libblkid_la)
 wipefs_CFLAGS = $(AM_CFLAGS) -I$(ul_libblkid_incdir)
 
-if BUILD_LSBLK
-bin_PROGRAMS += lsblk
-dist_man_MANS += lsblk.8
-lsblk_SOURCES = lsblk.c \
-		$(top_srcdir)/lib/canonicalize.c \
-		$(top_srcdir)/lib/ismounted.c \
-		$(top_srcdir)/lib/tt.c \
-		$(top_srcdir)/lib/strutils.c
-lsblk_LDADD = $(ul_libblkid_la)
-lsblk_CFLAGS = $(AM_CFLAGS) -I$(ul_libblkid_incdir)
-endif
-
 if HAVE_STATIC_BLKID
 sbin_PROGRAMS += blkid.static
 blkid_static_SOURCES = $(blkid_SOURCES)
diff --color -ru sys-utils/Makefile.am sys-utils/Makefile.am
--- util-linux-2.19.1/sys-utils/Makefile.am	2021-06-13 09:34:49.686370532 +1000
+++ util-linux-2.19.1/sys-utils/Makefile.am	2021-06-13 09:42:06.644523490 +1000
@@ -2,7 +2,7 @@
 
 bin_PROGRAMS =
 sbin_PROGRAMS =
-usrbin_exec_PROGRAMS = flock ipcrm ipcs ipcmk renice setsid
+usrbin_exec_PROGRAMS = ipcrm ipcs ipcmk renice setsid
 usrsbin_exec_PROGRAMS = readprofile
 
 dist_man_MANS = flock.1 ipcrm.1 ipcs.1 ipcmk.1 renice.1 setsid.1 \
@@ -10,8 +10,8 @@
 
 if LINUX
 bin_PROGRAMS += dmesg
-sbin_PROGRAMS += ctrlaltdel fsfreeze fstrim
-usrbin_exec_PROGRAMS += cytune setarch
+sbin_PROGRAMS += ctrlaltdel fsfreeze
+usrbin_exec_PROGRAMS += cytune
 usrsbin_exec_PROGRAMS += ldattach tunelp rtcwake
 
 dist_man_MANS += dmesg.1 ctrlaltdel.8 cytune.8 setarch.8 \
diff --color -ru text-utils/Makefile.am text-utils/Makefile.am
--- util-linux-2.19.1/text-utils/Makefile.am	2021-06-13 09:34:49.696370638 +1000
+++ util-linux-2.19.1/text-utils/Makefile.am	2021-06-13 09:46:19.230591066 +1000
@@ -2,7 +2,7 @@
 
 EXTRA_DIST = README.clear README.col
 
-usrbin_exec_PROGRAMS = col colcrt colrm column hexdump rev line tailf
+usrbin_exec_PROGRAMS = col colcrt colrm rev line tailf
 
 hexdump_SOURCES = hexdump.c conv.c display.c hexsyntax.c parse.c \
 		  hexdump.h
